lib/security/WSSecurity.js

```diff
@@ -3,10 +3,21 @@
 var crypto = require('crypto')
   , passwordDigest = require('../utils').passwordDigest;
 
-function WSSecurity(username, password, passwordType) {
+function WSSecurity(username, password, options) {
+  options = options || {};
   this._username = username;
   this._password = password;
-  this._passwordType = passwordType || 'PasswordText';
+  //must account for backward compatibility for passwordType String param as well as object options defaults: passwordType = 'PasswordText', hasTimeStamp = true   
+  if (typeof options === 'string') {
+    this._passwordType = options;
+    this._passwordType = this._passwordType === 'PasswordDigest' ? 'PasswordDigest' : 'PasswordText';
+  } else {
+    this._passwordType = typeof options.passwordType !== 'undefined' ? options.passwordType : 'PasswordText';
```

Comment:
@herom I understand your desire to shorten the code and make it more readable, unfortunately your suggested code does not pass the three new tests I have implemented.  Could I ask you to check out my commit and run the tests yourself to help me find a solution?